cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactCodec.java

```diff
@@ -118,16 +125,39 @@ public FactCodec(EntityTable entityTable, int resolution, int rollTimebaseInterv
     return rowKey;
   }
 
+  public byte[] getNextRowKey(List<TagValue> previousTags, byte[] rowKey) {
+    byte[] newRowKey = new byte[rowKey.length];
```

Comment:
Please add comment about the logic of it. I guess it is the following (please correct me, if I'm wrong):

``` java

// 0) result row key length is determined by the tagValues to be included, which is indexOfTagValueToChange plus one: the last one will be changing
// 1) the row key part up to the tagValue to be changed remains the same
// 2) to unchanged part we add incremented value of the key part at position of tagValue to be changed. We use Bytes.stopKeyForPrefix to increment that key part.
// 3) if key part cannot be incremented, then we return null, indicating "no next row key exist"

```